### DISCUSSION TITLE
Add User with Permission to Folder/Calendar

### DISCUSSION THREAD

**Discussion by lcnittl** [ecederstrand/exchangelib#1015] [Q&A]
I am trying to figure out how to grant another user (by e.g. `primary_smtp_address`) permission to a public folder/calendar to which I have owner rights (can grant permissions using Outlook).

I can query the current permissions with:

```py
>>> creds = exchangelib.Credentials("myusr", "mypwd")
>>> acc = exchangelib.Account("my.user@domain.tld", credentials=creds, autodiscover=True)
>>> calendar = acc.public_folders_root / "A Folder" / "A Calendar" 
>>> print(calendar.permission_set.calendar_permissions)
[CalendarPermission(can_create_items=True, can_create_subfolders=False, is_folder_owner=False, is_folder_visible=True, is_folder_contact=False, edit_items='Owned', delete_items='Owned', read_items='FullDetails', user_id=UserId(sid='<...>', primary_smtp_address='other.user@domain.tld', display_name='Firstname Surname', distinguished_user=None, external_user_identity=None), calendar_permission_level='Author'), CalendarPermission(<...>)]
```

But I am not quite sure on how to reverse this process.

Any help is highly appreciated! Thanks and cheers

**Comment by ecederstrand** [comment]
This would probably require support for the [AddDelegate](https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/adddelegate-operation) operation (and related UpdateDelegate and DeleteDelegate operations). exchangelib does not yet support these operations.
